MWPW-159202 Filter name is duplicated in custom link tracking for Catalog

bozojovicic opened the pull request:
There are 2 problems here :
- Bug 1 : On Catalog page, when you click different categories in the left side navigation couple of times, their names are just added to the existing event string making it longer after every click
<img width="1434" alt="catalog" src="https://github.com/user-attachments/assets/fec172cd-fc43-4f23-a498-2ca841bc3f30" />

- Bug 2 : With OOTB analytics system, the call `window._satellite.track` is done before our code sets the correct `daa-lh` values by `catalog.js` and then we have the analytics event strings like this `photo--cat|s2|video|nopzn|catalog|b2|filters` where 1st and 3rd strings point to 2 different categories. We need the 3rd string for click events on Desktop/Mobile/Web types where we have event strings like `Desktop--types|s2|photo|nopzn|catalog|b3|types` (3rd string points to the currently selected category)
For left side nav categories we now use `handleCustomAnalyticsEvent()` which is fired after all `daa-lh` attributes are correctly set.

Resolves: [MWPW-159202](https://jira.corp.adobe.com/browse/MWPW-159202)

Test page : `mwpw159202--cc--bozojovicic.aem.live/products/catalog`

**Test URLs:**
- Before: https://main--cc--adobecom.aem.live/?martech=off
- After: https://mwpw159202--cc--bozojovicic.aem.live/?martech=off

3ch023 reviewed (changes requested):
@bozojovicic nala would probably be too complicated, but could you add a unit test for it?

bozojovicic commented:
> @bozojovicic nala would probably be too complicated, but could you add a unit test for it?

@3ch023 we had some unit tests for analytics but not so well written so I fixed the existing tests and I added few more for the change I made in sidenav. A little problem is - analytics team never says how precisely the event string should look like so I guess sometimes we are sending junk and to make things even worse it seems that nobody is checking this on their side. We are sending analytics strings like `photo--cat|s2|social-mediasocial-mediasocial-mediaillustrationillustrationillustrationvideovideovideophotophotophotocreativity-designcreativity-designcreativity-designallall|b2|filters` for a long time and nobody noticed that on their side.

3ch023 commented:
> > @bozojovicic nala would probably be too complicated, but could you add a unit test for it?
> 
> @3ch023 we had some unit tests for analytics but not so well written so I fixed the existing tests and I added few more for the change I made in sidenav. A little problem is - analytics team never says how precisely the event string should look like so I guess sometimes we are sending junk and to make things even worse it seems that nobody is checking this on their side. We are sending analytics strings like `photo--cat|s2|social-mediasocial-mediasocial-mediaillustrationillustrationillustrationvideovideovideophotophotophotocreativity-designcreativity-designcreativity-designallall|b2|filters` for a long time and nobody noticed that on their side.

thank you!
yeah i am afraid they don't have monitored validation, so its better if we control it on our side..

Roycethan commented:
@bozojovicic Verified...Plz proceed with stage merge
